Add initial draft of Reactivity concept (#137)
solidjs/solid-docs-next: content/concepts/reactivity/index.mdx

}

function createEffect(fn) {
  // Add function as subscriber in global scope
  currentSubscriber = fn;
  // Run function to trigger the accessor of any signals that are called
  fn();
  // Remove the function as the current subscriber
  currentSubscriber = null;
}
```

This is all the code needed to create a basic reactive system. We can demonstrate that it works by incrementing the `count` value every second and watching the console.

```jsx
const [count, setCount] = createSignal(0);

createEffect(() => {
  console.log("The count is " + count());
});

setInterval(() => {
  setCount(count() + 1);
}, 1000);
```

<img
  style="max-width: 30rem; margin-bottom: 3rem; border-radius: 5px;"
  src={increment}
  alt="Incrementing counter"
/>

We have just implemented the most basic form of Solid's reactivity system!

## Effect tracking is synchronous

One observation we should note about our reactivity system is that it's synchronous. It registers the subscriber globally, runs the effect, and unregisters the subscriber.

So what happens if our `createEffect` function looks like this?

```jsx
createEffect(() => {
  setTimeout(() => {
    console.log(count());
  }, 1000);
});
```

> Our `createEffect` implementation doesn't wait around for this `setTimeout` callback to execute, so by the time we call our `count` accessor, there is no subscriber in the global scope. The `count` signal, therefore, will not register this callback as one of its subscribers!

Reviewer: ```suggestion
Our `createEffect` implementation doesn't wait around for this `setTimeout` callback to execute, so by the time we call our `count` getter, there is no subscriber in the global scope. The `count` signal won't register this callback as one of its subscribers.
```
